rust/src/graph/mod.rs

```diff
@@ -0,0 +1,137 @@
+use bimap::BiMap;
+use derive_new::new;
+use getset::{CopyGetters, Getters};
+use lazy_static::lazy_static;
+use rustc_hash::{FxHashMap, FxHashSet};
+use slotmap::{new_key_type, SecondaryMap, SlotMap};
+use std::sync::RwLock;
+use string_interner::backend::StringBackend;
+use string_interner::{DefaultSymbol, StringInterner};
+
+pub mod direct_import_queries;
+pub mod graph_manipulation;
+pub mod hierarchy_queries;
+pub mod higher_order_queries;
+pub mod import_chain_queries;
+
+pub(crate) mod pathfinding;
+
+lazy_static! {
+    static ref MODULE_NAMES: RwLock<StringInterner<StringBackend>> =
+        RwLock::new(StringInterner::default());
+    static ref EMPTY_MODULE_TOKENS: FxHashSet<ModuleToken> = FxHashSet::default();
+    static ref EMPTY_IMPORT_DETAILS: FxHashSet<ImportDetails> = FxHashSet::default();
+    static ref EMPTY_IMPORTS: FxHashSet<(ModuleToken, ModuleToken)> = FxHashSet::default();
+}
+
+new_key_type! { pub struct ModuleToken; }
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Getters, CopyGetters)]
+pub struct Module {
+    #[getset(get_copy = "pub")]
+    token: ModuleToken,
+
+    #[getset(get_copy = "pub")]
+    name: DefaultSymbol,
+
+    // Invisible modules exist in the hierarchy but haven't been explicitly added to the graph.
+    #[getset(get_copy = "pub")]
+    is_invisible: bool,
+
+    #[getset(get_copy = "pub")]
+    is_squashed: bool,
+}
+
+impl Module {
+    pub fn name_as_string(&self) -> String {
```

Comment:
Might be more ergonomic to call this `name` and call the field `interned_name`, `name_symbol` or something like that?

Reply:
Yes true your suggestion is much better 👍 Let's change it.